### PR TITLE
New hooks: `language.*:before|after`

### PR DESCRIPTION
**Pull request by afbora**
### 🎉 Features

- New language hooks: https://kirby.nolt.io/419
  - `language.create:before|after`: It is triggered when a new language is created.
  - `language.update:before|after`: It is triggered when the language is updated.
  - `language.delete:before|after`: It is triggered when the language is deleted.

### Breaking changes
None

## Ready?

- [x] Unit tests for fixed bug/feature
- [x] In-code documentation (wherever needed)
- [x] Tests and checks all pass

### For review team

- [x] Add changes to release notes draft in Notion
- [x] Add to [website docs release checklist](https://github.com/getkirby/getkirby.com/pulls) (if needed)

### CODE REVIEW

File: src/Cms/Language.php [hunk 1]
```diff
@@ -210,6 +210,9 @@ public static function create(array $props)
 			$props['default'] = true;
 		}
 
+		// trigger before hook
+		$kirby->trigger('language.create:before', compact('props'));
+
 		$language = new static($props);
```
Comment (bastianallgeier) [inline src/Cms/Language.php]: I think we should use the same parameters as we do in Page::create. 

```suggestion
		$language = new static($props);

		$kirby->trigger('language.create:before', ['language' => $language, 'input' => $props]);
```

File: src/Cms/Language.php [hunk 2]
```diff
@@ -224,6 +227,9 @@ public static function create(array $props)
 		// update the main languages collection in the app instance
 		App::instance()->languages(false)->append($language->code(), $language);
```
Comment (bastianallgeier) [inline src/Cms/Language.php]: We already have `$kirby` and we could clean this up. 

```suggestion
		$kirby->languages(false)->append($language->code(), $language);
```

File: src/Cms/Language.php [hunk 3]
```diff
@@ -224,6 +227,9 @@ public static function create(array $props)
...
 
+		// trigger after hook
+		$kirby->trigger('language.create:after', compact('language', 'props'));
```
Comment (bastianallgeier) [inline src/Cms/Language.php]: Same as above. It should get the language and the input – not props.

```suggestion
		$kirby->trigger('language.create:after', ['language' => $language, 'input' => $props]));
```

File: src/Cms/Language.php [hunk 4]
```diff
@@ -660,6 +676,12 @@ public function update(array $props = null)
 		// validate the updated language
 		LanguageRules::update($updated);
 
+		// trigger before hook
+		$kirby->trigger('language.update:before', [
+			'language' => $this,
+			'props' => $props
```
Comment (bastianallgeier) [inline src/Cms/Language.php]:
```suggestion
			'input' => $props
```

File: src/Cms/Language.php [hunk 5]
```diff
@@ -687,6 +709,13 @@ public function update(array $props = null)
 		// make sure the language is also updated in the Kirby language collection
 		App::instance()->languages(false)->set($language->code(), $language);
 
+		// trigger after hook
+		$kirby->trigger('language.update:after', [
+			'newLanguage' => $language,
+			'oldLanguage' => $this,
+			'props' => $props
```
Comment (bastianallgeier) [inline src/Cms/Language.php]:
```suggestion
			'input' => $props
```

File: src/Cms/Language.php [hunk 6]
```diff
@@ -225,10 +225,10 @@ public static function create(array $props)
 		}
 
 		// update the main languages collection in the app instance
-		App::instance()->languages(false)->append($language->code(), $language);
+		$kirby->languages(false)->append($language->code(), $language);
 
 		// trigger after hook
-		$kirby->trigger('language.create:after', compact('language', 'props'));
+		$kirby->trigger('language.create:after', ['language' => $language, 'input' => $props]));
```
Comment (bastianallgeier) [inline src/Cms/Language.php]: Yikes, this happens when I use the code suggestion feature here. There are now additional parenthesis. I guess it's easier to fix this locally.

Comment (afbora) [reply src/Cms/Language.php]: Actually I already fixed. This line should be outdated. Where do you found this?

Comment (bastianallgeier) [reply src/Cms/Language.php]: Oh great. Then it was just Github showing the outdated version.

**Review by bastianallgeier — APPROVED** [review]

**Pull request merged**